target/linux/ramips/dts/mt7620n_hootoo_tm05.dts

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620n.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "hootoo,tm05", "ralink,mt7620n-soc";
+	model = "HooToo TM05";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "tm05:blue:power";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wifi {
+			label = "tm05:green:wifi";
+			gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&i2c {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
```

Comment:
So, if we are certain there is nothing here (or at least, it is not the uboot environment data). Maybe this partition should be removed completely?

Merge it with "u-boot"?
In that case, you would simply change the length of u-boot partition like so
`reg = <0x0 0x40000>;`

@adrianschmutzler thoughts?

Reply:
FYI, there is something there, just not sure what it's for 😆. Here is the start of it,
```
root@OpenWrt:/# hexdump -C /dev/mtd1ro
00000000  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
00002000  b5 2b 34 45 43 6f 75 6e  74 72 79 52 65 67 69 6f  |.+4ECountryRegio|
00002010  6e 3d 30 00 43 6f 75 6e  74 72 79 52 65 67 69 6f  |n=0.CountryRegio|
00002020  6e 41 42 61 6e 64 3d 30  00 43 6f 75 6e 74 72 79  |nABand=0.Country|
00002030  43 6f 64 65 3d 55 53 00  42 73 73 69 64 4e 75 6d  |Code=US.BssidNum|
00002040  3d 31 00 73 73 69 64 5f  74 61 69 6c 3d 00 73 73  |=1.ssid_tail=.ss|
00002050  69 64 5f 70 72 65 3d 00  53 53 49 44 31 3d 54 72  |id_pre=.SSID1=Tr|
00002060  61 76 65 6c 4c 41 4e 00  53 53 49 44 32 3d 00 57  |avelLAN.SSID2=.W|
00002070  69 72 65 6c 65 73 73 4d  6f 64 65 3d 39 00 46 69  |irelessMode=9.Fi|
00002080  78 65 64 54 78 4d 6f 64  65 3d 48 54 00 54 78 52  |xedTxMode=HT.TxR|
00002090  61 74 65 3d 30 00 43 68  61 6e 6e 65 6c 3d 31 00  |ate=0.Channel=1.|
```

Reply:
@mpratt14 , this is `Config` in OEM firmware. Maybe default u-boot environment?! :thinking: 
I don't recommend to merge with `u-boot`!

Reply:
It was discovered that it is not the u-boot environment, and that it is actually at 0x1d0000 or something like that (the OEM partition called 'params'). Making it unusable without implementing the virtual_flash and a lot of nonsense.

My thinking is since it is not the u-boot environment, there is no reason Openwrt needs to know what it is or that it is separate from u-boot, so I would say it is just part of the OEM u-boot and was labeled incorrectly from the beginning.

Reply:
Please use "config" here. That also follows the convention in other devices.

Reply:
No issue here! Are others OK with this? Only asking because I copied this from other devices - I don't mind changing in, just want to make sure I'm not messing up some sort of "standard".

Thanks!

Reply:
No standard here, since we know its not the uboot env, I agree it should be named as it is in OEM and set to read only

Reply:
WFM, done!